### PR TITLE
[FIX] sms: fix sms button in phone field

### PR DESCRIPTION
**Pull request by SimonGenin**
The sms field extension had been somewhat badly reimplemented

### CODE REVIEW

File: addons/sms/static/src/components/sms_button/sms_button.js [hunk 1]
```diff
@@ -7,16 +7,33 @@ const { Component } = owl;
...
+                ...this.user.context,
+                default_res_model: this.props.record.resModel,
+                default_res_id: this.props.record.resId,
+                default_number_field_name: this.props.name,
+                default_composition_mode: 'comment',
+            }
+        }, {
+            onClose: async () => {
+                // reload chatter data
+                const messaging = await Component.env.services.messaging.get();
```
Comment (aab-odoo) [inline addons/sms/static/src/components/sms_button/sms_button.js:30]: this.props.record.model.load()
this.props.record.load({ resId }) ?

File: addons/sms/static/src/components/phone_field/phone_field.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <t t-inherit="web.PhoneField" t-inherit-mode="extension">
         <xpath expr="//*[hasclass('o_phone_link')]" position="after">
             <t t-if="props.value.length > 0">
-                <SendSMSButton record="props.record" value="props.value"/>
+                <SendSMSButton t-props="props" />
```
Comment (aab-odoo) [inline addons/sms/static/src/components/phone_field/phone_field.xml:7]: all props, or just what is necessary (resId, resModel, name, value)

Comment (SimonGenin) [reply addons/sms/static/src/components/phone_field/phone_field.xml:7]: I keep all props, cause as soon as we need resId and resModel, this is ORM linked so having them seperated doesn't bring any advantage

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**